lib/src/cpp/common/LocatorList.hpp

```diff
@@ -0,0 +1,368 @@
+ */
+void clear_port(
+        xercesc::DOMElement& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Remove locator physical port.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be modified.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is empty or not an integer.
+ */
+void clear_physical_port(
+        xercesc::DOMElement& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Remove locator IP address.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be modified.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is empty or not an integer.
+ */
+void clear_address(
+        xercesc::DOMElement& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Remove locator unique LAN ID.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be modified.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is empty or not an integer.
+ */
+void clear_unique_lan_id(
+        xercesc::DOMElement& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Remove locator WAN IP address.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be modified.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is empty or not an integer.
+ */
+void clear_wan_address(
+        xercesc::DOMElement& xml_node,
+        const std::string& index);
+
+/************************************************************************/
+/* Collection functions                                                 */
+/************************************************************************/
+
+/**
+ * @brief Append a locator with specified kind or update the existing locator kind.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] kind locator kind.
+ * @param[in] index Collection element to be changed. If empty, a new element is added to the list.
+ *
+ * @throw ElementInvalid Exception if the provided locator kind is not valid.
+ * @throw ElementNotFound Exception if index provided and the collection element is not found,
+ *        or if the element does not exist in the collection if an index is provided.
```

Comment:
```suggestion
 * @throw ElementNotFound Exception if the element does not exist in the collection if an index is provided.
```

I think that we can follow the same logic as in the clear function. The element existence require that the collection element exists.